### PR TITLE
Clear all filters button prop

### PR DESCRIPTION
**Pull request by khrizzcristian**
#### What problem is this solving?
I added a new prop on `filter-navigator.v3`. This prop will show a "Clear all" button if you have any filters selected on desktop devices.

[Workspace](https://longo--mcxhoreca.myvtex.com/LEGUME-SI-FRUCTE)

#### Screenshots or example usage:
Filters

### CODE REVIEW

**Comment by khrizzcristian** [comment]
I just added a new prop for the same component.

`"showSelectedFiltersInTitle"`. This property allows you to display the selected filters next to the filter title.

New Workspace for test:
https://lng--mcxhoreca.myvtex.com/LEGUME-SI-FRUCTE


Includes the both props.

selectedInTItle

**Comment by georgebrindeiro** [comment]
LGTM on the docs side

File: react/FilterNavigator.js [hunk 1]
```diff
@@ -159,6 +163,14 @@ const FilterNavigator = ({
...
   )
 
+  const hasFiltersApplied = window.location.href.split('?').length > 1
+
+  const handleResetFilters = () => {
+    const url = window.location.href
+
+    window.location.href = url.split('?')[0]
+  }
+
```
Comment (igorbrasileiro) [inline react/FilterNavigator.js:173]: Have you ever tried to use the render runtime hook?
https://github.com/vtex-apps/render-runtime/blob/1aa78341198fed752c4785aadce04e38d7ccdae5/react/components/RenderContext.tsx#L13

I don't ever use it, but I think it has these functionalities built-in.

**Review by igorbrasileiro — CHANGES_REQUESTED** [review]
Take a look on render runtime context, may help you

File: react/FilterNavigator.js (hunk 1 above)
Comment (khrizzcristian) [reply react/FilterNavigator.js:173]: Can you provide me an example please? How should I use this?

Comment (khrizzcristian) [reply react/FilterNavigator.js]: Fixed

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Added
+- `showClearAllFilters` prop to the `FilterNavigator`.
+- `showSelectedFiltersInTitle` prop to the `FilterNavigator`.
```
Comment (igorbrasileiro) [inline CHANGELOG.md:11]: Hey @klzns @victorhmp , would be good if only exists one line?
```suggestion
- `showClearAllFilters`, `showSelectedFiltersInTitle` props to the `FilterNavigator`.
```

Comment (victorhmp) [reply CHANGELOG.md]: Could be! No problem :)

**Review by hiagolcm — CHANGES_REQUESTED** [review]
We already have the `appliedFiltersOverview` prop
The `showSelectedFiltersInTitle` seems redundant. Since the`appliedFiltersOverview` follows the Baymard guide, I would suggest removing the `showSelectedFiltersInTitle`

image

**Comment by khrizzcristian** [comment]
> We already have the `appliedFiltersOverview` prop
> The `showSelectedFiltersInTitle` seems redundant. Since the`appliedFiltersOverview` follows the Baymard guide, I would suggest removing the `showSelectedFiltersInTitle`
> 
> image

I just add that prop:
https://longo--mcxhoreca.myvtex.com/Carne

This prop is good if the filters are used in sidebar ( see design ). There is a div role button with the title and dropdown arrow and another div with selectedFilters. There is no way to add that selected filters near title

**Comment by hiagolcm** [comment]
> I just add that prop:
> https://longo--mcxhoreca.myvtex.com/Carne
> 
> This prop is good if the filters are used in sidebar ( see design ). There is a div role button with the title and dropdown arrow and another div with selectedFilters. There is no way to add that selected filters near title

I understand, but two props that do the same thing except for the position still redundant. 

If you need this, I would suggest adding a new value to the appliedFiltersOverview. It accepts string values

**Comment by khrizzcristian** [comment]
showSelectedFiltersInTitle removed :)

**Comment by camilavcoutinho** [comment]
After reviewing locale files, we found a lot of translation issues, so I've created a PR (https://github.com/vtex-apps/search-result/pull/512) to fix the translations, and add "clear all" as well.

**Review by karenkrieger — APPROVED** [review]
README LGTM - Documentation approval

File: react/FilterNavigator.js [hunk 3]
```diff
@@ -159,6 +162,12 @@ const FilterNavigator = ({
     (facet) => facet.selected
   )
 
+  const hasFiltersApplied = window.location.href.split('?').length > 1
```
Comment (hiagolcm) [inline react/FilterNavigator.js:165]: This is not a good way to check the applied filters. It will give you false positives since any query string will trigger this.
I think you can use the `useSearchPage` to get the selectedFilters

File: react/FilterNavigator.js [hunk 4]
```diff
@@ -263,6 +272,21 @@ const FilterNavigator = ({
...
               priceRangeLayout={priceRangeLayout}
             />
+            {showClearAllFilters && hasFiltersApplied && (
+              <div
+                className={`${applyModifiers(
+                  handles.filter__container,
+                  'clearFilters'
+                )} bb b--muted-4`}
+              >
+                <button
```
Comment (hiagolcm) [inline react/FilterNavigator.js:282]: Could you try to use the Button component from styleguide?
`import { Button } from 'vtex.styleguide'`

File: react/FilterNavigator.js [hunk 5]
```diff
@@ -86,6 +88,7 @@ const FilterNavigator = ({
   navigationTypeOnMobile = 'page',
   updateOnFilterSelectionOnMobile = false,
   showClearByFilter = false,
+  showClearAllFilters = false,
```
Comment (hiagolcm) [inline react/FilterNavigator.js:91]: Since it is only on the desktop, I think we should make it clear.
Maybe we should change the prop name to `showClearAllFiltersOnDesktop`

**Review by hiagolcm — CHANGES_REQUESTED** [review]
It's looking good. I commented on some suggestions.
Also, I want to test the reset function with some scenarios. Could you recreate the workspace, please?

**Comment by khrizzcristian** [comment]
New workspace: 

https://clearall--mcxhoreca.myvtex.com/Brutarie-si-patiserie

**Comment by khrizzcristian** [comment]
Any update for this? :)

File: messages/da.json [hunk 6]
```diff
@@ -52,6 +52,7 @@
   "store/search-result.filter-breadcrumbs.primary": "Filtre",
   "store/search-result.filter-button.apply": "Anvend",
   "store/search-result.filter-button.clear": "Ryd",
+  "store/search-result.filter-button.clearAll": "Ryd",
```
Comment (camilavcoutinho) [inline messages/da.json:55]:
```suggestion
  "store/search-result.filter-button.clearAll": "Ryd alle",
```

File: messages/el.json [hunk 7]
```diff
@@ -52,6 +52,7 @@
   "store/search-result.filter-breadcrumbs.primary": "Φίλτρα",
   "store/search-result.filter-button.apply": "Εφαρμογή",
   "store/search-result.filter-button.clear": "Καθαρισμός",
+  "store/search-result.filter-button.clearAll": "Καθαρισμός",
```
Comment (camilavcoutinho) [inline messages/el.json:55]:
```suggestion
  "store/search-result.filter-button.clearAll": "Απαλοιφή Όλων",
```

File: messages/fi.json [hunk 8]
```diff
@@ -52,6 +52,7 @@
   "store/search-result.filter-breadcrumbs.primary": "Suodattimet",
   "store/search-result.filter-button.apply": "Käytä",
   "store/search-result.filter-button.clear": "Puhdista",
+  "store/search-result.filter-button.clearAll": "Puhdista",
```
Comment (camilavcoutinho) [inline messages/fi.json:55]:
```suggestion
  "store/search-result.filter-button.clearAll": "Tyhjennä kaikki",
```

File: messages/pl.json [hunk 9]
```diff
@@ -52,6 +52,7 @@
   "store/search-result.filter-breadcrumbs.primary": "Filtry",
   "store/search-result.filter-button.apply": "Zastosuj",
   "store/search-result.filter-button.clear": "Wyczyść",
+  "store/search-result.filter-button.clearAll": "Wyczyść",
```
Comment (camilavcoutinho) [inline messages/pl.json:55]:
```suggestion
  "store/search-result.filter-button.clearAll": "Wyczyść wszystko",
```

File: messages/ru.json [hunk 10]
```diff
@@ -52,6 +52,7 @@
   "store/search-result.filter-breadcrumbs.primary": "Фильтры",
   "store/search-result.filter-button.apply": "Применить",
   "store/search-result.filter-button.clear": "Очистить",
+  "store/search-result.filter-button.clearAll": "Очистить",
```
Comment (camilavcoutinho) [inline messages/ru.json:55]:
```suggestion
  "store/search-result.filter-button.clearAll": "Очистить все",
```

File: messages/sk.json [hunk 11]
```diff
@@ -52,6 +52,7 @@
   "store/search-result.filter-breadcrumbs.primary": "Filtre",
   "store/search-result.filter-button.apply": "Použiť",
   "store/search-result.filter-button.clear": "Vyčistiť",
+  "store/search-result.filter-button.clearAll": "Vyčistiť",
```
Comment (camilavcoutinho) [inline messages/sk.json:55]:
```suggestion
  "store/search-result.filter-button.clearAll": "Vymazať všetko",
```

File: messages/sl.json [hunk 12]
```diff
@@ -52,6 +52,7 @@
   "store/search-result.filter-breadcrumbs.primary": "Filtri",
   "store/search-result.filter-button.apply": "Uporabi",
   "store/search-result.filter-button.clear": "Počisti",
+  "store/search-result.filter-button.clearAll": "Počisti",
```
Comment (camilavcoutinho) [inline messages/sl.json:55]:
```suggestion
  "store/search-result.filter-button.clearAll": "Počisti vse",
```

File: messages/uk.json [hunk 13]
```diff
@@ -52,6 +52,7 @@
   "store/search-result.filter-breadcrumbs.primary": "Фільтри",
   "store/search-result.filter-button.apply": "Застосувати",
   "store/search-result.filter-button.clear": "Очистити",
+  "store/search-result.filter-button.clearAll": "Очистити",
```
Comment (camilavcoutinho) [inline messages/uk.json:55]:
```suggestion
  "store/search-result.filter-button.clearAll": "Очистити все",
```

File: messages/ca.json [hunk 14]
```diff
@@ -52,6 +52,7 @@
   "store/search-result.filter-breadcrumbs.primary": "Filtres",
   "store/search-result.filter-button.apply": "Aplica",
   "store/search-result.filter-button.clear": "Esborra",
+  "store/search-result.filter-button.clearAll": "Esborra",
```
Comment (camilavcoutinho) [inline messages/ca.json:55]:
```suggestion
  "store/search-result.filter-button.clearAll": "Esborra-ho tot",
```

File: messages/bg.json [hunk 15]
```diff
@@ -52,6 +52,7 @@
   "store/search-result.filter-breadcrumbs.primary": "Филтри",
   "store/search-result.filter-button.apply": "Прилагане",
   "store/search-result.filter-button.clear": "Изчистване",
+  "store/search-result.filter-button.clearAll": "Изчистване",
```
Comment (camilavcoutinho) [inline messages/bg.json:55]:
```suggestion
  "store/search-result.filter-button.clearAll": "Изчистване на всички",
```

File: messages/fr.json [hunk 16]
```diff
@@ -52,6 +52,7 @@
   "store/search-result.filter-breadcrumbs.primary": "Filtres",
   "store/search-result.filter-button.apply": "Faire la demande",
   "store/search-result.filter-button.clear": "Nettoyer",
+  "store/search-result.filter-button.clearAll": "Nettoyer",
```
Comment (camilavcoutinho) [inline messages/fr.json:55]:
```suggestion
  "store/search-result.filter-button.clear": "Effacer",
  "store/search-result.filter-button.clearAll": "Tout effacer",
```

File: messages/ko.json [hunk 17]
```diff
@@ -52,6 +52,7 @@
   "store/search-result.filter-breadcrumbs.primary": "필터",
   "store/search-result.filter-button.apply": "적용",
   "store/search-result.filter-button.clear": "깨끗한",
+  "store/search-result.filter-button.clearAll": "깨끗한",
```
Comment (camilavcoutinho) [inline messages/ko.json:55]:
```suggestion
  "store/search-result.filter-button.clearAll": "모두 지우기",
```

File: messages/nl.json [hunk 18]
```diff
@@ -52,6 +52,7 @@
   "store/search-result.filter-breadcrumbs.primary": "Filters",
   "store/search-result.filter-button.apply": "Toepassen",
   "store/search-result.filter-button.clear": "Zuiver",
+  "store/search-result.filter-button.clearAll": "Zuiver",
```
Comment (camilavcoutinho) [inline messages/nl.json:55]:
```suggestion
  "store/search-result.filter-button.clearAll": "Alles wissen",
```

File: messages/ro.json [hunk 19]
```diff
@@ -52,6 +52,7 @@
   "store/search-result.filter-breadcrumbs.primary": "Filtre",
   "store/search-result.filter-button.apply": "Aplică",
   "store/search-result.filter-button.clear": "Curățare",
+  "store/search-result.filter-button.clearAll": "Resetează toate",
```
Comment (camilavcoutinho) [inline messages/ro.json:55]:
```suggestion
  "store/search-result.filter-button.clearAll": "Șterge tot",
```

File: messages/sv.json [hunk 20]
```diff
@@ -52,6 +52,7 @@
   "store/search-result.filter-breadcrumbs.primary": "Filter",
   "store/search-result.filter-button.apply": "Tillämpa",
   "store/search-result.filter-button.clear": "Rengör",
+  "store/search-result.filter-button.clearAll": "Rengör",
```
Comment (camilavcoutinho) [inline messages/sv.json:55]:
```suggestion
  "store/search-result.filter-button.clearAll": "Rensa alla",
```

File: messages/es.json [hunk 21]
```diff
@@ -53,6 +53,7 @@
   "store/search-result.filter-breadcrumbs.primary": "Filtros",
   "store/search-result.filter-button.apply": "Aplicar",
   "store/search-result.filter-button.clear": "Limpiar",
+  "store/search-result.filter-button.clearAll": "Limpiar",
```
Comment (camilavcoutinho) [inline messages/es.json:56]:
```suggestion
  "store/search-result.filter-button.clearAll": "Limpiar todo",
```

File: messages/pt.json [hunk 22]
```diff
@@ -53,6 +53,7 @@
   "store/search-result.filter-breadcrumbs.primary": "Filtros",
   "store/search-result.filter-button.apply": "Aplicar",
   "store/search-result.filter-button.clear": "Limpar",
+  "store/search-result.filter-button.clearAll": "Limpar",
```
Comment (camilavcoutinho) [inline messages/pt.json:56]:
```suggestion
  "store/search-result.filter-button.clearAll": "Limpar tudo",
```

File: messages/de.json [hunk 23]
```diff
@@ -52,6 +52,7 @@
   "store/search-result.filter-breadcrumbs.primary": "Filter",
   "store/search-result.filter-button.apply": "Anwenden",
   "store/search-result.filter-button.clear": "Sauber",
+  "store/search-result.filter-button.clearAll": "Sauber",
```
Comment (camilavcoutinho) [inline messages/de.json:55]:
```suggestion
  "store/search-result.filter-button.clear": "Löschen",
  "store/search-result.filter-button.clearAll": "Alles löschen",
```

**Review by camilavcoutinho — CHANGES_REQUESTED** [review]

**Review by camilavcoutinho — APPROVED** [review]
LGTM

**Review by igorbrasileiro — APPROVED** [review]
I don't have enough context about it, but the code looks good to me.

**Comment by igorbrasileiro** [comment]
> It's looking good. I commented on some suggestions.
> Also, I want to test the reset function with some scenarios. Could you recreate the workspace, please?

Is everything ok? Can you review it again?

File: react/FilterNavigator.js [hunk 24]
```diff
@@ -159,6 +164,14 @@ const FilterNavigator = ({
     (facet) => facet.selected
   )
 
+  const { searchQuery } = useSearchPage()
+  const hasFiltersApplied = searchQuery.variables.selectedFacets.length > 1
+
+  const { history, navigate } = useRuntime()
+  const handleResetFilters = () => {
+    navigate({ to: history.location.pathname })
```
Comment (hiagolcm) [inline react/FilterNavigator.js:172]: This function removes any querystring from the URL and this is a problem. There are some querystrings  that still needed after cleaning the filters

Ac5lOt2Om6

In this case for example. The proper URL should be:

 https://hiago--storecomponents.myvtex.com/top?_q=top&fuzzy=0&map=ft&operator=and&query=/top&searchState

But the button is redirecting to:

https://hiago--storecomponents.myvtex.com/top

In other words, this function is removing any querystring from the URL, even querystrings that are not related to the filters at all

I would recommend using the `navigateToFacet` function. This function will do all the work of setting the proper URL. You just need to pass all the selected facets. I think it is gonna solve the problem

**Review by hiagolcm — CHANGES_REQUESTED** [review]

File: react/FilterNavigator.js (hunk 24 above)
Comment (khrizzcristian) [reply react/FilterNavigator.js:172]: Hello, How can i use that function? I can't figured it out how to use it

Comment (hiagolcm) [reply react/FilterNavigator.js:172]: the function is decleared here: https://github.com/vtex-apps/search-result/blob/6d1e7666c07ad5d521c553c41dccceb78f3c632d/react/FilterNavigator.js#L163

and here is how we use it to clean the filters on mobile: 
https://github.com/vtex-apps/search-result/blob/ce253793d668ff84af8bb47330ff7c9ddd1c8271/react/components/FilterSidebar.js#L143

Comment (khrizzcristian) [reply react/FilterNavigator.js]: https://clearall--mcxhoreca.myvtex.com/brutarie-si-patiserie?_q=Brutarie-si-patiserie&map=category-1,brand,temperatura-minima,cantitate-pachet&query=/brutarie-si-patiserie/la-lorraine/-18/40&searchState

**Review by hiagolcm — APPROVED** [review]
After the adjustments, it's working for the scenarios I had complained about. LGTM 🚀

**Review by thalytafabrine — APPROVED** [review]

File: messages/ro.json [hunk 25]
```diff
@@ -52,6 +52,7 @@
   "store/search-result.filter-breadcrumbs.primary": "Filtre",
   "store/search-result.filter-button.apply": "Aplică",
   "store/search-result.filter-button.clear": "Curățare",
```
Comment (dvm1984) [inline messages/ro.json:54]: Șterge

**Comment by dvm1984** [comment]
please replace curatare with Șterge

File: messages/ro.json (hunk 25 above)
Comment (khrizzcristian) [reply messages/ro.json:54]: That translation is not part of my PR

**Comment by vtexdaniel** [comment]
@hiagolcm, what will happen with this PR? Does the partner need to review it?
